README.md

```diff
@@ -12,6 +12,8 @@ Download files from any public URL to `/data/out/files`.
 
 - `baseUrl` (required) -- common part of URL
 - `path` (required) -- path part of URL (futureproof to allow row configs)
+- `client_options` (optional)
+  - `max_redirects` (optional) -- maximum number of redirects to follow
```

Comment:
čekáš nějaký další `client_options`? nezatěžujeme end usera zbytečnejma implementačníma detailama? nemohl by ten parametr bejt na root levelu? i kdyby tam toho bylo víc...

Reply:
Minimálně někdy asi přibude timeout pro pomalejší servery (nebo naopak pokud to chceme rychle a nechceme čekat 30s), no-verify pro případy, kdy taháš něco z nějakého jejich serveru se špatným HTTPS, http-auth, atp. Ale asi máš pravdu, že to zbytečně odhaluje co je uvnitř. Zanoření umožnilo to elegantně celé předat dovnitř jako jeden parametr. Na druhou stranu bych v UI tyhle věci zabalil stejně do nějakého "bloku", třeba Advanced nebo Extended settings a nezobrazoval bych je přímo vedle base+path.

Reply:
Jasně, v UI to může bejt poskládaný do bloků, ale ty vůbec nemusej souviset s tím, kde to je v configu uložený. Zanoření v `parameters` jsme tu u novějších komponent ještě neměli, tak jsem byl zvědavej :-) Jestli to nikomu jinýmu nevadí, tak bych to takhle nechal.

Reply:
db extractory a writery to maji taky zanoreny

Reply:
Ja to sklonim trochu jinak.
Nelibi se mi jmena parametru. Jednou je to camelCase a podruhe s '_'.

Melo by to byt jednotne a u vseho noveho se snazime tlacit camelCase

Reply:
Kludne to dajme flat. t.j. vedla seba `baseUrl`, `path` a `maxRedirect` ... ako sa to namapuje interne, to je asi jedno a bude sa stym lepsie robit v UI. A takisto aj ked to budem konfigurovat manualne

Reply:
ty jo o tom camel case jsem nevedel, v tom pripade by mel byt i tady https://github.com/keboola/app-snowflake-dwh-manager/blob/efcb0b5ea2d857d9d52245140e50ffbcb4e52c73/src/ConfigDefinition.php#L22

Reply:
v komponetnách taky dávám všude snake_case (a v UI camelCase)

@odinuv db writery a extraktory nejsou "novější" :-) a tam to dává smysl, credentials apod! už mlčím!

Reply:
@ondrejhlavacek koukam na tvuj s3 extractor a bigquery writer, zadnej snake case tam nevidim.

Reply:
V examplech dokumentace je taky camelCase https://developers.keboola.com/extend/component/ui-options/configuration-schema/

Reply:
@ErikZigo Nojo! :-( byl jsem myšlenkama u input/output mappingu, kde to teda fakt je snake_case, a processor-create-manifest taky. čéče, dobrej postřeh, měli bychom se pro jedno rozhodnout a držet. možná ten snake_case dělám jen já! 💀

Reply:
no rekl bych, ze ho delas jen ty :D, tak ona to imho neni zas takova tragedie mezi komponentama, ale urcite v jedne komponente stejny, takze kdyz je tu  `baseUrl` tak je blbost mit `client_options`

Reply:
super, máme viníka :-)

myslíte, že bych to měl celý přebouchat, aby bylo všude camelCase? může to bejt celkem snadno zpětně kompatibilní.

Reply:
j taky bych na to přepouchání kašlal.

Reply:
Ideální by bylo to přidat do code stylu a kontrolovat automaticky. Co není automatické, to se neuhlídá :)

Reply:
resp. muselo by to být v phpstanu - protože by to potřebovalo naparsovat ConfigDefinition a asi by to  bylo netriviální... :/

Reply:
Každopádně to jdu přepsat.